app/controllers/schools_controller.rb

```diff
@@ -0,0 +1,39 @@
+class SchoolsController < ApplicationController
+  def show
+    @school = School.find(params[:id])
+  end
+
+  def new; end
+
+  def create
+    validate_opened_date
+    return if performed?
+
+    @school = School.new(params.require(:school).permit(:name, :opened, :school_type))
+
+    if @school&.save
+      redirect_to @school
+    else
+      render :new
+    end
+  end
+
+private
+
+  def validate_opened_date
```

Comment:
Definitely better to define a custom validation in the model. It looks a bit out of place in here.

Reply:
I tried that, but couldn't quite get it, since the date seemed to be cast before model validation ran. How would you do that?

Reply:
Missed this yesterday. I'll take a look this morning or perhaps we can pair via google meets or something